components/suggest/src/yelp.rs

```diff
@@ -0,0 +1,322 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+use interrupt_support::SqlInterruptScope;
+use rusqlite::types::ToSqlOutput;
+use rusqlite::{named_params, Connection, Result as RusqliteResult, ToSql};
+use sql_support::ConnExt;
+use std::sync::Mutex;
+use url::form_urlencoded;
+
+use crate::{
+    provider::SuggestionProvider, rs::DownloadedYelpDataset, suggestion::Suggestion, Result,
+    SuggestionQuery,
+};
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+#[repr(u8)]
+enum Modifier {
+    PreModifier = 0,
+    PostModifier = 1,
+}
+
+impl ToSql for Modifier {
+    fn to_sql(&self) -> RusqliteResult<ToSqlOutput<'_>> {
+        Ok(ToSqlOutput::from(*self as u8))
+    }
+}
+
+/// max subject: 50 + pre-modifier: 10 + post-modifier: 10 + location-sign: 7 + location: 50 = 127 = 150.
```

Comment:
Could you please briefly outline the query strategy here? A link to the doc for details also works.